pkg/apis/eventing/v1alpha1/implements_test.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2018 The Knative Authors
+ Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/knative/pkg/apis/duck"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+)
+
+func TestTypesImplements(t *testing.T) {
+	var emptyGen duckv1alpha1.Generation
+	testCases := []struct {
+		instance interface{}
+		iface    duck.Implementable
+	}{
+		// Channel
+		{instance: &Channel{}, iface: &duckv1alpha1.Conditions{}},
```

Comment:
IMO, it will be coherent if we add the tests for each type under their respective test files. e.g `channel_types_test.go` for `Channel`.

Reply:
Mainly did it the same as on `pkg` following https://github.com/knative/pkg/pull/98#discussion_r220566483 — but i don't really have strong opinion on that :angel:

Reply:
@neosab @vaikas-google should I update the PR or we can do that on a follow-up ? :angel:

Reply:
Unless @neosab feels strongly, I'm fine with getting this in as is.

Reply:
Sorry folks, I was on vacation 🍹  last week so missed this. I would have been fine :)